tests/by-util/test_expr.rs

```diff
@@ -123,6 +123,21 @@ fn test_or() {
         .args(&["-14", "|", "1"])
         .succeeds()
         .stdout_only("-14\n");
+
+    new_ucmd!()
+        .args(&["1", "|", "a", "/", "5"])
+        .succeeds()
+        .stdout_only("1\n");
+    
+    new_ucmd!()
+        .args(&["foo", "|", "a", "/", "5"])
+        .succeeds()
+        .stdout_only("foo\n");
+
+    new_ucmd!()
+        .args(&["0", "|", "10", "/", "5"])
+        .succeeds()
+        .stdout_only("2\n");
```

Comment:
do you know why the result is 2 here?

Reply:
if the first value is 0, which means `value_as_bool(&value) == false`,
then the code i have modified will executed as below:
```rust
if op_type == "|" {
    if let Some(value) = operands.next() {
        let value = value.evaluate()?;
        out.push(value.clone());
    }
}
```
just do the same thing as origin code (`next` do the same thing as `for`), and will not effect the evaluate of `10 / 5`.
GNU produce the same result:
```shell
$ expr 0 \| 10 / 5
2
```

Reply:
ok, thanks :)